ipn/ipnlocal/serve.go

```diff
@@ -750,14 +750,22 @@ func (b *LocalBackend) serveWebHandler(w http.ResponseWriter, r *http.Request) {
 	http.Error(w, "empty handler", 500)
 }
 
+// stripMountPointFromError removes the root directory from the error message
+// to avoid exposing the local filesystem path.
+func stripMountPointFromError(err error, fileOrDir string) string {
+	errMsg := err.Error()
+	return strings.Replace(errMsg, fileOrDir, "./", 1)
+}
+
 func (b *LocalBackend) serveFileOrDirectory(w http.ResponseWriter, r *http.Request, fileOrDir, mountPoint string) {
 	fi, err := os.Stat(fileOrDir)
 	if err != nil {
 		if os.IsNotExist(err) {
 			http.NotFound(w, r)
 			return
 		}
-		http.Error(w, err.Error(), 500)
+		sanitizedErr := stripMountPointFromError(err, fileOrDir)
+		http.Error(w, sanitizedErr, 500)
```

Comment:
I think we shouldn't be writing these errors directly to the client anyway. Instead, we can construct a friendly error using https://pkg.go.dev/path/filepath#Rel and log the real error for debugging